api/core/tools/provider/builtin/json_process/tools/insert.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Union
+from core.tools.entities.tool_entities import ToolInvokeMessage
+from core.tools.tool.builtin_tool import BuiltinTool
+
+import json
+from jsonpath_ng import parse
+
+class JSONParseTool(BuiltinTool):
+    def _invoke(self,
+                user_id: str,
+                tool_parameters: dict[str, Any],
+                ) -> Union[ToolInvokeMessage, list[ToolInvokeMessage]]:
+        """
+            invoke tools
+        """
+        # get content
+        content = tool_parameters.get('content', '')
+        if not content:
+            return self.create_text_message('Invalid parameter content')
+        
+        # get query
+        query = tool_parameters.get('query', '')
+        if not query:
+            return self.create_text_message('Invalid parameter query')
+        
+        # get new value
+        new_value = tool_parameters.get('new_value', '')
+        if not new_value:
+            return self.create_text_message('Invalid parameter new_value')
+        
+        # get insert position
+        index = tool_parameters.get('index', None)
+        
+        # get create path
+        create_path = tool_parameters.get('create_path', False)
+        
+        try:
+            result = self._insert(content, query, new_value, index, create_path)
+            return self.create_text_message(str(result))
+        except Exception:
+            return self.create_text_message('Failed to insert JSON content')
+
+
+    def _insert(self, origin_json, query, new_value, index=None, create_path=False):
+        try:
+            input_data = json.loads(origin_json)
+            expr = parse(query)
+            try:
+                new_value = json.loads(new_value)
+            except json.JSONDecodeError:
+                new_value = new_value
+            
+            matches = expr.find(input_data)
+            
+            if not matches and create_path:
+                # 创建新路径
```

Comment:
Pls use English comments instead, so that it could make sense for global developers

Reply:
Thank you for your suggestion. I have replace the comments in English.